azurerm/internal/services/compute/windows_virtual_machine_scale_set_resource.go

```diff
@@ -233,9 +233,9 @@ func resourceArmWindowsVirtualMachineScaleSet() *schema.Resource {
 				ForceNew: true,
 				Default:  string(compute.Manual),
 				ValidateFunc: validation.StringInSlice([]string{
-					string(compute.Automatic),
+					string(compute.AutomaticByOS),
 					string(compute.Manual),
-					string(compute.Rolling),
+					string(compute.AutomaticByPlatform),
```

Comment:
Should be `compute.UpgradeModeAutomatic`, `compute.UpgradeModeManual` and `compute.UpgradeModeRolling`